lib/board.rb

```diff
@@ -21,8 +21,68 @@ def initialize
                "D4" => Cell.new("D4")
               }
   end
+  # looping through - nested loop for expansion
 
   def valid_coordinate?(coordinate)
     cells.keys.include?(coordinate)
   end
+
+  def all_same?(array)
+    array.uniq.length == 1 
+  end
+
+  def not_all_same?(array)
+    array.uniq.length > 1 
+  end
+
+  def valid_placement?(ship, cells)
+    letters = cells.map do |cell| 
+      cell.split('').first
+    end
+    #should we convert the letters to integers here instead of at the end?
+
+    numbers = cells.map do |cell|
+      cell.split('').last.to_i
+    end
+    # Could this be a case expression? 
+    # TDD by the books?
+    if ship.length != cells.length
+      false 
+
+    elsif not_all_same?(letters) && not_all_same?(numbers)
+      false 
+
+    elsif all_same?(letters) && not_all_same?(numbers) 
+
+      numbers.each_cons(2).all? do |num_1, num_2| 
```

Comment:
This definitely needs to be a helper method. Maybe we call it `consecutive?`

Reply:
I like that!